libs/StatusGoQt/src/StatusGo/Chat/ChatDto.h

```diff
@@ -0,0 +1,132 @@
+#pragma once
+
+#include <nlohmann/json.hpp>
+
+#include <vector>
+
+#include <QColor>
+
+using json = nlohmann::json;
+
+namespace Status::StatusGo::Chats {
+
+    constexpr auto ChannelGroupTypeUnknown = "unknown";
+    constexpr auto ChannelGroupTypePersonal = "personal";
+    constexpr auto ChannelGroupTypeCommunity = "community";
+
+    enum ChatType
+    {
+        Unknown = 0,
+        OneToOne = 1,
+        Public = 2,
+        PrivateGroupChat = 3,
+        Profile = 4,
+        CommunityChat = 6
+    };
+
+    struct Category {
+        QString id;
+        QString name;
+        int position;
+    };
+
+    struct Permission {
+        int access;
+        bool ensOnly;
+    };
+
+    struct Images {
+        QString thumbnail;
+        QString large;
+        QString banner;
+    };
+
+    struct ChatMember {
+        QString id;
+        bool admin;
+        bool joined;
+        std::vector<int> roles;
+    };
+
+    struct ChatDto {
```

Comment:
Why DTO? Usually what is not a DTO should be inferred from its name (SomeService is a service, `AccountFactory` is a factory ...). for me `Chat` is a DTO or an entity so it is not adding meaningful information to it. I also find that adding functionality close to this data representation is the best place.

Reply:
@stefandunca `Chat` is too general in this app, also for me there is a big difference between `Chat` and `ChatDto`, cause `ChatDto` refers to the chat DataTransferObject, the object which we're receiving from remote (in our case `status-go`) and chat refers to the object which is mapped in the shape/need we want to use it in the desktop app. `ChatDto` has minimal set of data we're keeping, but `Chat` will have some additional flags/props resolved on the desktop app side.